mlos_bench/mlos_bench/optimizers/mlos_core_optimizer.py

```diff
@@ -111,6 +111,8 @@ def bulk_register(self, configs: Sequence[dict], scores: Sequence[Optional[float
             df_configs = df_configs[df_status_completed]
             df_scores = df_scores[df_status_completed]
         self._opt.register(df_configs, df_scores)
+        if not is_warm_up:
+            self._iter += len(df_scores)
```

Comment:
Does `bulk_register()` get called outside of the warmup procedure?  Or is that for resuming an experiment?

Reply:
That will be pretty dependent on timing though, right?
We may want some notion of a group commit to wait for others to come in.
For instance, config_a score comes in a 10 seconds, config_b score comes in at 10.1 seconds, but we've already collected the first one and started retraining.  We could wait for a full batch to finish, or just some timeline (e.g., 500ms) for a collection to come in before retraining and starting a new one.

Anyways ... future work.